tests/phpunit/includes/Core/Modules/Module_With_Owner_ContractTests.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Module_With_Owner_ContractTests
+ *
+ * @package   Google\Site_Kit\Tests\Core\Modules
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Site_Kit\Core\Modules\Module_With_Owner;
+use Google\Site_Kit\Core\Modules\Module_With_Settings;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\Storage\Setting;
+use Google\Site_Kit\Core\Storage\Setting_With_Owned_Keys_Interface;
+use Google\Site_Kit\Tests\TestCase_Context_Trait;
+
+trait Module_With_Owner_ContractTests {
+
+	use TestCase_Context_Trait;
+
+	/**
+	 * @return Module_With_Owner
+	 */
+	abstract protected function get_module_with_owner();
+
+	public function test_owner_id() {
+		$testcase = $this->get_testcase();
+		$module   = $this->get_module_with_owner();
+
+		// By default ownerID is 0.
+		$testcase->assertEquals( 0, $module->get_owner_id() );
+
+		if ( $module instanceof Module_With_Settings ) {
+			$settings = $module->get_settings();
+
+			if ( $settings instanceof Setting ) {
+				$settings->register();
+			}
+
+			if ( $settings instanceof Setting_With_Owned_Keys_Interface ) {
```

Comment:
I realize the `instanceof` checks make the IDE happy here, but these contract tests should cause test failures if used for testing modules that don't implement these things. That's part of the benefit of these tests as they can enforce aspects of the contract which are not possible to do via PHP alone.

So rather than using `instanceof` with an `if` and nesting, let's simply assert `$testcase->assertInstanceOf( Module_With_Settings::class, $module );`. We can assume `get_settings` returns an instance of `Setting` since it returns a `Module_Settings`. Similarly, we can assert `$testcase->assertInstanceOf( Setting_With_Owned_Keys_Interface::class, $settings );` rather than nesting there as well, and now we have a nice flat method.

Again, it's not just about formatting, but that these tests should fail if the types are not implemented correctly, either because the test was used on a test case where it shouldn't be, or because something was done improperly and needs to be fixed for the test to pass 🙂

Reply:
Updated.